desk/app/activity.hoon

```diff
@@ -696,10 +697,39 @@
 ++  del-source
   |=  =source:a
   ^+  cor
+  =.  cor
+    =/  children  (get-children:src indices source)
+    |-
+    ?~  children  cor
+    =.  cor  (del-source i.children)
+    $(children t.children)
   =.  indices  (~(del by indices) source)
+  =.  activity  (~(del by activity) source)
   =.  volume-settings  (~(del by volume-settings) source)
   ::  TODO: send notification removals?
   (give-update [%del source] [%hose ~])
+++  del-event
+  |=  [=source:a event=incoming-event:a]
+  ^+  cor
+  =/  =index:a  (~(got by indices) source)
+  =/  events=(list [=time-id:a =event:a])
+    %+  murn
+      (tap:on-event:a stream.index)
+    |=  [=time e=event:a]
+    ?.  =(-.e event)  ~
+    `[time e]
+  ?~  events  cor
+  =/  new-index  index(stream +:(del:on-event:a stream.index time-id:(head events)))
+  =.  indices
+    (~(put by indices) source new-index)
+  =.  cor  (refresh source)
+  =/  new-activity=activity:a
+    %+  roll
+      (snoc (get-parents:src source) source)
+    |=  [=source:a out=activity:a]
+    (~(put by out) source (~(gut by activity) source *activity-summary:a))
```

Comment:
Can this not be a `got:by`? We _just_ refreshed this source, and so we also refreshed its parents, right?